src/gcovr/data_model/coverage.py

```diff
@@ -1517,7 +1523,13 @@ def has_reportable_calls(self) -> bool:
 
     def calls(self) -> Iterable[CallCoverage]:
         """Iterate over the calls."""
-        yield from [v for _, v in sorted(self.__calls.items())]
+        yield from [
+            v
+            for _, v in sorted(
+                self.__calls.items(),
+                key=lambda kv: tuple(x if x is not None else -1 for x in kv[0]),
+            )
+        ]
```

Comment:
Use a sentinel value to avoid

```
  File "/home/julien/Virtualenvs/py312/lib/python3.12/site-packages/gcovr/data_model/coverage.py", line 1520, in calls
    yield from [v for _, v in sorted(self.__calls.items())]
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: '<' not supported between instances of 'int' and 'NoneType'
```